crawler/rss.py

```diff
@@ -16,26 +16,37 @@ def preprocess(cls, source, options):
         return [RSSConfig.links[source]]
     
     @classmethod
-    def extract(cls, objs, options):
+    def extract(cls, objs, code, options):
         """
         Extract news from BeautifulSoup object.
         """
         # TODO Optimize to each newspaper style
         soup = objs[0]
-        ret = list()
-        for entry in soup.findAll("entry"):
+        ret = list()      
+
+        for entry in soup.findAll(RSSConfig.item_name[code]):
             temp = dict()
+            temp["category"] = list()
             for item in entry.children:
-                # print(item.name)
-                if item.name == None: 
+                if item.name == None:   
+                    continue
+                elif item.name == RSSConfig.item_publish[code]:
                     continue
+                elif item.name == "title":
+                    temp[item.name] = item.string
+                elif item.name == RSSConfig.item_content[code]:
+                    temp["content"] = item.get_text().strip()
                 elif item.name == "link":
-                    temp["link"] = item["href"]
-                elif item.name == "author":
+                    # Separate with and without href attribute 
+                    if "href" in item:
+                        temp["link"] = item["href"]
+                    else:
+                        temp[item.name] = item.string
```

Comment:
What is in item.name and item.string? I don't understand the comment on line 40.